addons/godot-next/references/property_info.gd

```diff
@@ -0,0 +1,62 @@
+# author: xdgamestudios
+# license: MIT
+# description: A class abstraction, for building properties to be rendered to the
+#              inspector using the method _get_property_list()
+tool
+extends Reference
+class_name PropertyInfo
+
+##### CONSTANTS #####
+
+const SELF_PATH := "res://addons/godot-next/references/property_info.gd";
+
+##### PROPERTIES #####
+
+var name: String;
+var type: int;
+var hint: int;
+var hint_string: String;
+var usage: int;
+
+##### NOTIFICATIONS #####
+
+func _init(p_name: String = "", p_type: int = TYPE_NIL, p_hint: int = PROPERTY_HINT_NONE, p_hint_string: String = "", p_usage: int = PROPERTY_USAGE_DEFAULT):
+	name = p_name;
+	type = p_type;
+	hint = p_hint;
+	hint_string = p_hint_string;
+	usage = p_usage;
+
+##### PUBLIC METHODS #####
+
+func to_dict() -> Dictionary:
+	return {
+		"name": name,
+		"type": type,
+		"hint": hint,
+		"hint_string": hint_string,
+		"usage": usage
+	}
+
+static func from_dict(p_dict: Dictionary) -> PropertyInfo:
+	var name = p_dict.name;
```

Comment:
This should likewise have a ternary if-else like the others to cover cases where the supplied Dictionary doesn't have a `name` or `type` field. It wouldn't make sense for them to NOT have them, but you also don't want to allow for the possibility of a fatal error if someone neglects to do it for some reason.